integration-test/shadow/shadow_system_test.c

```diff
@@ -659,6 +668,144 @@ static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateDelta, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/documents with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateDocuments, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* First of all, try to delete any Shadow document in the cloud.
+     * This could trigger the /delete/accepted or /delete/rejected
+     * based on the thing status on the cloud.
+     */
+    ( void ) createTopicString( ShadowTopicStringTypeDelete, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    topicString,
+                                                    updateDocument,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /delete/accepted or /delete/rejected. */
+    TEST_ASSERT_TRUE( ( receivedDeleteAcceptedResult || receivedDeleteRejectedResult ) );
+
+    /* Publish to the shadow topic /update with reported payload,
+     *  that we subscribed to, with Qos 0. */
+    ( void ) createTopicString( ShadowTopicStringTypeUpdate, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    topicString,
+                                                    TEST_SHADOW_DESIRED,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /update/documents*/
+    TEST_ASSERT_TRUE( receivedUpdateDocumentsResult );
+
+    /* Check the flag for /update/delta. */
+    TEST_ASSERT_TRUE( receivedUpdateDeltaResult );
+
+    /* Check the flag for /update/accepted and /update/rejected. */
+    TEST_ASSERT_TRUE( ( receivedUpdateAcceptedResult || receivedUpdateRejectedResult ) );
+
+    /* Finally, sending null payload on topic /get to trigger /get/accepted. */
+    ( void ) createTopicString( ShadowTopicStringTypeGet, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    topicString,
+                                                    updateDocument,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /get/accepted and /get/rejected. */
+    TEST_ASSERT_TRUE( ( receivedGetAcceptedResult || receivedGetRejectedResult ) );
```

Comment:
It will also be better to know if these incoming publishes are received for a classic shadow vs named shadow.

Reply:
Now in eventCallback() I am checking that both the Thing name and Shadow name are correct before setting the Result flags.